src/Atmos/Model/boundaryconditions.jl

```diff
@@ -37,6 +37,102 @@ Set the value at the boundary to match the `init_state!` function. This is mainl
 """
 struct InitStateBC <: BoundaryCondition
 end
-function atmos_boundarycondition!(bc::InitStateBC, m::AtmosModel, stateP::Vars, diffP::Vars, auxP::Vars, nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t) 
+function atmos_boundarycondition!(bc::InitStateBC, m::AtmosModel, stateP::Vars, diffP::Vars, auxP::Vars, nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t,_...) 
   init_state!(m, stateP, auxP, auxP.coord, t)
 end
+
+
+"""
+  DYCOMS_BC <: BoundaryCondition
+  Prescribes boundary conditions for Dynamics of Marine Stratocumulus Case
+"""
+struct DYCOMS_BC{DT} <: BoundaryCondition
+  C_drag::DT
+  LHF::DT
+  SHF::DT
+end
+function atmos_boundarycondition!(bc::DYCOMS_BC, m::AtmosModel, stateP::Vars, diffP::Vars, auxP::Vars, nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, state1::Vars, diff1::Vars, aux1::Vars) 
+  # stateM is the 𝐘⁻ state while stateP is the 𝐘⁺ state at an interface. 
+  # at the boundaries the ⁻, minus side states are the interior values
+  # state1 is 𝐘 at the first interior nodes relative to the bottom wall 
+  DT = eltype(stateP)
+  # Get values from minus-side state
+  ρM = stateM.ρ 
+  UM, VM, WM = stateM.ρu
+  EM = stateM.ρe
+  QTM = stateM.moisture.ρq_tot
+  uM, vM, wM  = UM/ρM, VM/ρM, WM/ρM
+  q_totM = QTM/ρM
+  UnM = nM[1] * UM + nM[2] * VM + nM[3] * WM
+  
+  # Assign reflection wall boundaries (top wall)
+  stateP.ρu = SVector(UM - 2 * nM[1] * UnM, 
+                      VM - 2 * nM[2] * UnM,
+                      WM - 2 * nM[3] * UnM)
+
+  # Assign scalar values at the boundaries 
+  stateP.ρ = ρM
+  stateP.moisture.ρq_tot = QTM
+  # Assign diffusive fluxes at boundaries
+  diffP = diffM
+  xvert = auxM.coord[3]
+  
+  if bctype == 5 # bctype identifies bottom wall 
```

Comment:
using numbers to identify a boundary may lead to errors. Can we wrap those numbers by explicit naming such as "front, back, east, west, top, bottom" or something along these lines?